aws/cloudformation/bootstrap_chef_stack.sh.erb

```diff
@@ -76,14 +76,21 @@ CHAIN=$(aws s3 cp $CURRENT/chain.pem - | replace_newline)
 KEY=$(aws s3 cp $CURRENT/key.pem - | replace_newline)
 <% end -%>
 
-<% if database && !rack_env?(:production) -%>
 apt-get install -y  jq
+
+<% if database && !rack_env?(:production) -%>
 DB_SECRET_NAME=${DatabaseSecretAdmin}
 DB_SECRET=$(aws --region $REGION secretsmanager get-secret-value --secret-id $DB_SECRET_NAME --version-stage AWSCURRENT | jq -r ".SecretString")
 DB_USERNAME=$(echo $DB_SECRET | jq -r ".username")
 DB_PASSWORD=$(echo $DB_SECRET | jq -r ".password")
 <% end -%>
 
+<% if rack_env?(:adhoc) -%>
+# Chef managed environments (staging, test, levelbuilder, production) authenticate to GitHub with an SSH key.
+# We're not using curly braces to expand ENVIRONMENT for complex reasons (this script is processed by CloudFormation !Sub).
+GITHUB_TOKEN=$(aws --region $REGION secretsmanager get-secret-value --secret-id "$ENVIRONMENT/cdo/github_token" --version-stage AWSCURRENT | jq -r ".SecretString")
+<% end -%>
```

Comment:
Is this env-specific gating necessary, or can we have our logic apply only when the `cdo/github_token` secret is set for the current environment?

Reply:
I only created an AWS Secret for adhocs (`adhoc/cdo/github_token`). I had previously created Secrets for all environment types, but then realized that we provision an SSH key for `staging`, `levelbuilder, `test`, and `production` that they use to authenticate content commit and push operations. To avoid complex bash logic that tests for a non-empty Secret being retrieved, I fugred gating by environment types = `adhoc` would be simplest.